modules/common/src/main/java/com/sygic/modules/common/poi/manager/PoiDataManager.kt

```diff
@@ -0,0 +1,59 @@
+package com.sygic.modules.common.poi.manager
+
+import androidx.annotation.RestrictTo
+import com.sygic.sdk.map.`object`.ViewObject
+import com.sygic.sdk.places.LocationInfo
+import com.sygic.sdk.places.Place
+import com.sygic.sdk.places.Places
+import com.sygic.sdk.position.GeoCoordinates
+import com.sygic.sdk.search.ReverseGeocoder
+import com.sygic.sdk.search.ReverseSearchResult
+import com.sygic.ui.common.sdk.data.PoiData
+import com.sygic.ui.common.sdk.extension.getFirst
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+interface PoiDataManager {
+
+    abstract class Callback : Places.PlaceListener, ReverseGeocoder.ReverseSearchResultsListener {
+        abstract fun onDataLoaded(poiData: PoiData)
+
+        final override fun onPlaceLoaded(place: Place) {
+            onDataLoaded(
+                PoiData(
```

Comment:
toto bude chciet asi builder pri takom mnozstve parametrov :)

Reply:
Však on to je defakto builder, protože má v kotlinu default parametry a pro Javu je tam todo, jak jsme se bavili ;)